pkgs/development/libraries/openmpi/default.nix

```diff
@@ -40,21 +52,28 @@ in stdenv.mkDerivation rec {
     find -name "Makefile.in" -exec sed -i "s/\`date\`/$ts/" \{} \;
   '';
 
-  buildInputs = with stdenv; [ gfortran zlib ]
+  buildInputs = with stdenv; [ gfortran zlib libfabric pmix ucx ]
     ++ lib.optionals isLinux [ libnl numactl ]
     ++ lib.optionals cudaSupport [ cudatoolkit ]
     ++ [ libevent hwloc ]
     ++ lib.optional (isLinux || isFreeBSD) rdma-core;
 
   nativeBuildInputs = [ perl ];
 
-  configureFlags = with stdenv; lib.optional (!cudaSupport) "--disable-mca-dso"
+  configureFlags = with stdenv; [
+    "--with-cma"
+    "--with-hwloc=${hwloc.dev}"
+    "--with-ofi=${libfabric}"
+    "--with-pmi=${pmix}"
+    "--with-pmix=${pmix} --enable-install-libpmix"
+    "--with-verbs=${rdma-core}"
+    "--with-ucx=${ucx}"
```

Comment:
We (@mekeetsa) are testing UCX with infiniband at the moment (see also https://github.com/markuskowa/NixOS-QChem/blob/ucx/ucx/default.nix). Did you test it with an Infiniband configuration?

Reply:
yes, i've tested it and it seems to work (setting up the env with `OMPI_MCA_btl=^openib,uct,tcp` `UCX_TLS=ib,sm,self`)